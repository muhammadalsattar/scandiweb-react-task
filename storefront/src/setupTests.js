const currencies = [
    {
        label: 'USD',
        symbol: '$',
    },
    {
        label: 'GBP',
        symbol: 'Â£',
    },
];
const categories = [
    {
        name: 'category1',
    },
    {
        name: 'category2',
    },
];
const attributes = [
    {
        id: 'Size',
        name: 'Size',
        type: 'text',
        items: [
            {
                id: 40,
                value: '40',
                displayValue: '40',
            },
        ],
    },
    {
        id: 'Color',
        name: 'Color',
        type: 'swatch',
        items: [
            {
                id: 'red',
                value: 'red',
                displayValue: 'Red',
            },
        ],
    },
];
const gallery = [
    'https://dummyimage.com/600x400/000/fff',
    'https://dummyimage.com/600x400/000/ggg',
    'https://dummyimage.com/600x400/000/hhh',
];
const prices = [
    {
        amount: 518.47,
        currency: currencies[0],
    },
    {
        amount: 372.67,
        currency: currencies[1],
    },
];
const products = [
    {
        id: '1',
        name: 'product1',
        brand: 'brand1',
        attributes,
        description: '<p>dummy text</p>',
        gallery,
        prices,
        category: categories[0],
        inStock: true,
        quantity: 1,
        selectedAttr: [{ name: 'Size', value: '40' }],
    },
    {
        id: '2',
        name: 'product2',
        brand: 'brand2',
        attributes,
        description: '<p>dummy text</p>',
        gallery,
        prices,
        category: categories[1],
        inStock: false,
        quantity: 1,
        selectedAttr: [{ name: 'Color', value: 'red' }],
    },
    {
        id: '3',
        name: 'product3',
        brand: 'brand3',
        attributes,
        description: '<p>dummy text</p>',
        gallery,
        prices,
        category: categories[0],
        inStock: false,
        quantity: 1,
        selectedAttr: [{ name: 'Size', value: '41' }],
    },
    {
        id: '4',
        name: 'product4',
        brand: 'brand4',
        attributes,
        description: '<p>dummy text</p>',
        gallery,
        prices,
        category: categories[1],
        inStock: false,
        quantity: 1,
        selectedAttr: [{ name: 'Color', value: 'orange' }],
    },
    {
        id: '5',
        name: 'product5',
        brand: 'brand5',
        attributes,
        description: '<p>dummy text</p>',
        gallery,
        prices,
        category: categories[0],
        inStock: true,
        quantity: 1,
        selectedAttr: [{ name: 'Size', value: '42' }],
    },
    {
        id: '6',
        name: 'product6',
        brand: 'brand6',
        attributes,
        description: '<p>dummy text</p>',
        gallery,
        prices,
        category: categories[1],
        inStock: false,
        quantity: 1,
        selectedAttr: [{ name: 'Color', value: 'green' }],
    },
];

export const initialState = {
    currencies,
    categories,
    products,
};
