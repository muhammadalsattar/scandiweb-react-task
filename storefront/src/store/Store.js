import { configureStore, combineReducers } from '@reduxjs/toolkit';
import cartReducer from '../reducers/cartReducer';
import productsReducer from '../reducers/productsReducer';
import categoriesReducer from '../reducers/categoriesReducer';
import currencyReducer from '../reducers/currencyReducer';

const rootReducer = combineReducers({
    cart: cartReducer,
    products: productsReducer,
    categories: categoriesReducer,
    currencies: currencyReducer,
});
// creating store and combining reducers
const Store = configureStore({
    reducer: rootReducer,
});

export default Store;
