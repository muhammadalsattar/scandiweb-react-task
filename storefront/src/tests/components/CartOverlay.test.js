import React from "react";
import {render, screen} from "@testing-library/react";
import { Provider } from "react-redux";
import {BrowserRouter as Router} from "react-router-dom";
import Store from "../../store/Store.js";
import {CartOverlay} from "../../components/Cartoverlay.js";
import { initialState } from "../../setupTests.js"

test("Cartoverlay renders correctly", () => {
    const { undefined, container } = render(
        <Provider store={Store}>
        <Router><CartOverlay cart={initialState.cart} defaultCurrency={initialState.currencies[0]}/></Router>
        </Provider>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText(/1178.01/i)).toBeInTheDocument();
    expect(screen.getByText(/View Bag/i)).toBeInTheDocument();
});