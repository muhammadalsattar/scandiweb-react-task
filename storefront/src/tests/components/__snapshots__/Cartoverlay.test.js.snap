// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cartoverlay renders correctly 1`] = `
<div>
  <div
    class="cart-overlay"
    hidden=""
  >
    <p>
      <b>
        My Bag, 
      </b>
      2
       items
    </p>
    <div
      class="cart-item"
    >
      <div
        class="details"
      >
        <p
          class="brand"
        >
          Sony
        </p>
        <p
          class="name"
        >
          PlayStation 5
        </p>
        <p>
          <b>
            $
            844.02
          </b>
        </p>
        <div
          class="cart-item-attribute"
        >
          <p
            class="attribute-name"
          >
            Color
            :
          </p>
          <div
            class="attribute-value"
          >
            <div
              class="Color selected"
            >
              <button
                style="background-color: rgb(68, 255, 3);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(3, 255, 247);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(3, 11, 255);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(0, 0, 0);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(255, 255, 255);"
              />
            </div>
          </div>
        </div>
        <div
          class="cart-item-attribute"
        >
          <p
            class="attribute-name"
          >
            Capacity
            :
          </p>
          <div
            class="attribute-value"
          >
            <div
              class="other-attr selected"
            >
              <button>
                512G
              </button>
            </div>
            <div
              class="other-attr"
            >
              <button>
                1T
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="quantity-image"
      >
        <div
          class="quantity"
        >
          <button>
            +
          </button>
          <p>
            <b>
              1
            </b>
          </p>
          <button>
            -
          </button>
        </div>
        <div
          class="image"
        >
          <img
            alt="PlayStation 5"
            src="https://images-na.ssl-images-amazon.com/images/I/510VSJ9mWDL._SL1262_.jpg"
            style="width: 50px;"
          />
        </div>
      </div>
    </div>
    <div
      class="cart-item"
    >
      <div
        class="details"
      >
        <p
          class="brand"
        >
          Microsoft
        </p>
        <p
          class="name"
        >
          Xbox Series S 512GB
        </p>
        <p>
          <b>
            $
            333.99
          </b>
        </p>
        <div
          class="cart-item-attribute"
        >
          <p
            class="attribute-name"
          >
            Color
            :
          </p>
          <div
            class="attribute-value"
          >
            <div
              class="Color selected"
            >
              <button
                style="background-color: rgb(68, 255, 3);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(3, 255, 247);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(3, 11, 255);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(0, 0, 0);"
              />
            </div>
            <div
              class="Color"
            >
              <button
                style="background-color: rgb(255, 255, 255);"
              />
            </div>
          </div>
        </div>
        <div
          class="cart-item-attribute"
        >
          <p
            class="attribute-name"
          >
            Capacity
            :
          </p>
          <div
            class="attribute-value"
          >
            <div
              class="other-attr selected"
            >
              <button>
                512G
              </button>
            </div>
            <div
              class="other-attr"
            >
              <button>
                1T
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="quantity-image"
      >
        <div
          class="quantity"
        >
          <button>
            +
          </button>
          <p>
            <b>
              1
            </b>
          </p>
          <button>
            -
          </button>
        </div>
        <div
          class="image"
        >
          <img
            alt="Xbox Series S 512GB"
            src="https://images-na.ssl-images-amazon.com/images/I/71vPCX0bS-L._SL1500_.jpg"
            style="width: 50px;"
          />
        </div>
      </div>
    </div>
    <div
      class="cart-total"
    >
      <p>
        Total
      </p>
      <p>
        $
        1178.01
      </p>
    </div>
    <div
      class="bag-checkout"
    >
      <div
        class="view-bag"
      >
        <a
          href="/cart"
        >
          <button>
            View Bag
          </button>
        </a>
      </div>
      <div
        class="checkout"
      >
        <button>
          Check out
        </button>
      </div>
    </div>
  </div>
</div>
`;
